bindings/python/pymongoarrow/schema.py

```diff
@@ -17,26 +17,29 @@
 
 
 class Schema:
-    """A mapping of field names to data types."""
-    def __init__(self, schema):
-        """Create a :class:`~pymongoarrow.schema.Schema` instance from a
-        mapping or an iterable.
+    """A mapping of field names to data types.
+
+    To create a schema, provide its constructor a mapping or an iterable
+    containing field names and their expected types, e.g.::
 
-        If ``schema`` is a mapping, each key must be a field name and the
-        corresponding value must be the expected data type of the named field.
-        If ``schema`` is an iterable, it must be comprised entirely of 2-member
-        sub-iterables. The first member of each sub-iterable must be a field
-        name and the second value must be the corresponding data type.
+       schema1 = Schema({'field_1': int, 'field_2': float})
+       schema2 = Schema([('field_1', int), ('field_2', float)])
```

Comment:
Why do we support an iterable here instead of only a mapping?

Reply:
🤷🏽 . Should we not?

Reply:
It makes it harder to reason about types. Eg `def __init__(self, schema: Mapping[str, Any]):` is simple, `def __init__(self, schema: Union[Mapping[str, Any],Iterable[Tuple[str, Any]]]):` is much more complicated to read.

Reply:
I would voted to remove it since it doesn't seem to provide any value.

Reply:
Note if you choose to do this we should do it in a separate ticket. This PR LGTM.

Reply:
Ok sounds good... I agree, we can remove it.